torch/csrc/jit/passes/frozen_ops_to_mkldnn.cpp

```diff
@@ -271,6 +273,32 @@ Operation createUnaryOp(
   };
 }
 
+void MKLDNNLayerNormOp(Stack* stack) {
+  c10::impl::ExcludeDispatchKeyGuard edkg(c10::autograd_dispatch_keyset);
+
+  // enable_cudnn not used
+  pop(stack);
+  auto eps = pop(stack).toDouble();
+
+  Tensor bias{};
+  Tensor weight{};
+  auto bias_ival = pop(stack);
+  TORCH_INTERNAL_ASSERT(bias_ival.isTensor());
+  bias = bias_ival.toTensor();
+
+  auto weight_ival = pop(stack);
+  TORCH_INTERNAL_ASSERT(weight_ival.isTensor());
+  weight = weight_ival.toTensor();
+
+  auto shape = pop(stack).toIntVector();
+  auto input = pop(stack).toTensor();
+
+  at::Tensor dst, mean, rstd;
+  std::tie(dst, mean, rstd) =
```

Comment:
so mean and variance here in the layer norm docs says: 
`Mean and variance are computed at runtime or provided by a user`. Could we provide this instead of computing on every run, is that possible? is it correct to initialize to zero ?

> For forward propagation, the mean and variance might be either computed at runtime (in which case they are outputs of the primitive) or provided by a user (in which case they are inputs). In the latter case, a user must set the dnnl_use_global_stats flag. 

are we sure we're doing this right ?

Reply:
> Could we provide this instead of computing on every run, is that possible?

mean and variance are computed for the input tensor, though, not sure how we can avoid re-computation on every run unless they are computed somewhere else in a graph? 

> are we sure we're doing this right ? 

We aren't setting the flag and we are recomputing them, so we should be using those tensors solely as outputs.

Reply:
yea, we could try caching them for inference. that doesnt need to be done in the first version of this pr but it might be a nice freezing speed up. the mean and variance are wrt the weights right ?

Reply:
unfortunately, it's wrt inputs :(